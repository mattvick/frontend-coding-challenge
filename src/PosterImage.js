import React from 'react';
import PropTypes from 'prop-types';

const PosterImage = ({ baseUrl, size, path, alt }) => <img src={`${baseUrl}${size}${path}`} alt={alt} />;

PosterImage.defaultProps = {
  baseUrl: '//image.tmdb.org/t/p/',
  size: 'w92',
  alt: 'Movie poster image',
};

PosterImage.propTypes = {
  baseUrl: PropTypes.oneOf([
    '//image.tmdb.org/t/p/',
    'https://image.tmdb.org/t/p/',
    'http://image.tmdb.org/t/p/',
  ]),
  size: PropTypes.oneOf(['w92', 'w154', 'w185', 'w342', 'w500', 'w780', 'original']),
  alt: PropTypes.string,
  path: PropTypes.string.isRequired,
};

export default PosterImage;
