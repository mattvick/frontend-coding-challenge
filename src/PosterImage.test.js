/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import PosterImage from './PosterImage';

const path = '/d5iIlFn5s0ImszYzBPb8JPIfbXD.jpg';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders with or without alt text', () => {
  act(() => {
    render(<PosterImage path={path} />, container);
  });
  expect(container.querySelector('img').getAttribute('alt'))
    .toBe('Movie poster image');

  const alt = 'Pulp Fiction movie poster';
  act(() => {
    render(<PosterImage path={path} alt={alt} />, container);
  });
  expect(container.querySelector('img').getAttribute('alt'))
    .toBe(alt);
});

it('renders without baseUrl or size', () => {
  act(() => {
    render(<PosterImage path={path} />, container);
  });
  expect(container.querySelector('img').getAttribute('src'))
    .toBe(`//image.tmdb.org/t/p/w92${path}`);
});

it('renders with size', () => {
  const size = 'w500';
  act(() => {
    render(<PosterImage path={path} size={size} />, container);
  });
  expect(container.querySelector('img').getAttribute('src'))
    .toBe(`//image.tmdb.org/t/p/${size}${path}`);
});

it('renders with baseUrl', () => {
  const baseUrl = 'http://image.tmdb.org/t/p/';
  act(() => {
    render(<PosterImage path={path} baseUrl={baseUrl} />, container);
  });
  expect(container.querySelector('img').getAttribute('src'))
    .toBe(`${baseUrl}w92${path}`);
});

test('PosterImage matches snapshot', () => {
  const component = renderer.create(
    <PosterImage path={path} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
