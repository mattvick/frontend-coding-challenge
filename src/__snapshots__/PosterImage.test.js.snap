// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterImage matches snapshot 1`] = `
<img
  alt="Movie poster image"
  src="//image.tmdb.org/t/p/w92/d5iIlFn5s0ImszYzBPb8JPIfbXD.jpg"
/>
`;
